### PR TITLE
Move Conditional Logic from .pubxml to Project Files

### PR DESCRIPTION
**Pull request by ShashankNay**
The conditional logic for `PublishReadyToRun` and `PublishTrimmed` properties was moved from the `.pubxml` files to the project file, ensuring proper configuration for Debug and Release builds regardless of the platform.

### CODE REVIEW

File: dev/VSIX/ProjectTemplates/Desktop/CSharp/UnitTestApp/WinUI.Desktop.Cs.UnitTestApp.csproj [hunk 1]
```diff
@@ -44,6 +44,13 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
+  <!-- Publish Properties -->
+  <PropertyGroup>
+    <PublishReadyToRun Condition="'$(Configuration)' == 'Debug'">False</PublishReadyToRun>
+    <PublishReadyToRun Condition="'$(Configuration)' != 'Debug'">True</PublishReadyToRun>
+    <PublishTrimmed Condition="'$(Configuration)' == 'Debug'">False</PublishTrimmed>
```
Comment (Scottj1s) [inline dev/VSIX/ProjectTemplates/Desktop/CSharp/UnitTestApp/WinUI.Desktop.Cs.UnitTestApp.csproj:51]: probably need to maintain the unconditional setting of PublishTrimmed=false.  I think Keith needed that for unit tests, which have lots of 'unreferenced' code.

Comment (ShashankNay) [reply dev/VSIX/ProjectTemplates/Desktop/CSharp/UnitTestApp/WinUI.Desktop.Cs.UnitTestApp.csproj:51]: Thanks, I wondered whether or not that was intentional

**Review by Scottj1s — APPROVED** [review]
thanks!

**Comment by ghost1372** [comment]
@ShashankNay @Scottj1s  why removed Conditional Logic added into `WinUI.Desktop.Cs.SingleProjectPackagedApp.csproj
` and not in `ProjectTemplate.csproj`?

**Comment by eduardobragaxz** [comment]
@Scottj1s You guys could put the condition on the PropertyGroup and then have only one of each property, no need to specify the False ones.

```
<PropertyGroup Condition="'$(Configuration)' != 'Debug'">

<PublishReadyToRun>True</PublishReadyToRun>
<PublishTrimmed>True</PublishTrimmed>

</PropertyGroup>
```

**Pull request merged**